docs/contributions.md

```diff
@@ -11,21 +11,37 @@ The following document is a rule set of guidelines for contributing.
 ## Table of Contents <!-- omit from toc -->
 
 - [Contributions](#contributions)
+  - [Before you start](#before-you-start)
   - [Code Contributions](#code-contributions)
-    - [Step 1: Fork](#step-1-fork)
-    - [Step 2: Branch](#step-2-branch)
-    - [Step 3: Commit](#step-3-commit)
-    - [Step 4: Sync](#step-4-sync)
-    - [Step 5: Push](#step-5-push)
-    - [Step 6: Pull Request](#step-6-pull-request)
+      - [Step 1: Fork](#step-1-fork)
+      - [Step 2: Branch](#step-2-branch)
+      - [Step 3: Commit](#step-3-commit)
+        - [Generating a GPG key](#generating-a-gpg-key)
+        - [Configure Git](#configure-git)
+        - [Commit Format](#commit-format)
+      - [Step 4: Sync](#step-4-sync)
+      - [Step 5: Push](#step-5-push)
+      - [Step 6: Pull Request](#step-6-pull-request)
   - [Commit message guidelines](#commit-message-guidelines)
   - [Coding guidelines](#coding-guidelines)
-    - [Code structuring patterns](#code-structuring-patterns)
-      - [Object Oriented Programming](#object-oriented-programming)
-      - [Functional Programming](#functional-programming)
-    - [Naming patterns](#naming-patterns)
-    - [Documentation](#documentation)
-    - [Writing tests](#writing-tests)
+      - [Code structuring patterns](#code-structuring-patterns)
+          - [Object Oriented Programming](#object-oriented-programming)
+          - [Functional Programming](#functional-programming)
+      - [Naming patterns](#naming-patterns)
+      - [Documentation](#documentation)
+      - [Writing tests](#writing-tests)
+
+## Before you start
+
+The best way to contribute to IF is to build your own plugins. You can build these in your own Github repositories or even publish them as packages, and then run them through IF without ever having to deal with IF core source code. 
+
+However, if you want to contribute changes to our repositories, either `if`, `if-docs`, `if-plugins` or `if-unofficial-plugins` the best way to start s to check the issue boards. 
```

Comment:
The typo in the "the best way to start s" text `i` is missing

Reply:
```suggestion
However, if you want to contribute changes to our repositories, either `if`, `if-docs`, `if-plugins` or `if-unofficial-plugins` the best way to start is to check the issue boards. 
```